src/components/vr_cooperation/src/vr_module.cc

```diff
@@ -363,5 +366,34 @@ void VRModule::ResetDefaultService() {
   default_app_id_ = 0;
 }
 
+void VRModule::RegisterService(int32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (std::find(registered_apps_.begin(), registered_apps_.end(), app_id)
+      == registered_apps_.end()) {
+    registered_apps_.push_back(app_id);
+  }
+}
+
+void VRModule::UnregisterService(int32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::vector<int32_t>::iterator it = registered_apps_.begin();
+  for (it = registered_apps_.begin(); it != registered_apps_.end(); ++it) {
```

Comment:
std::vector::iterator it = std::find(vector_of_registered_apps.begin(), vector_of_registered_apps.end(), app_id);
if (vector_of_registered_apps.end() != it) {
vector_of_registered_apps.erase(it);
}